src/net/gateway-archive.ts

```diff
@@ -0,0 +1,151 @@
+import { FileApi } from "../api/file";
+import { GatewayArchiveError } from "../errors/gateway-archive";
+import { TextRecordKeys } from "../models/entity"
+import { VochainProcessStatus } from "../models/protobuf";
+import { getEnsPublicResolverByNetwork } from "../util/ens";
+import { ContentUri } from "../wrappers/content-uri"
+import { EthNetworkID, IGatewayClient, IGatewayDVoteClient } from "../common";
+
+export interface IArchiveResponseBody {
+    process?: {
+        [k: string]: any
+    }
+    results?: {
+        [k: string]: any
+    },
+}
+
+export namespace GatewayArchive {
+
+    /**
+     * Fetch the full state of the given processId on the process archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getProcessStateArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
```

Comment:
Isn't the "archive" fragment of the name redundant?
It would be used like `GatewayArchive.getProcessStateArchive()`
=> `getProcessState` ?